php-frontend/src/main/java/org/sonar/php/parser/PHPGrammar.java

```diff
@@ -706,6 +709,11 @@ public static void declaration(LexerlessGrammarBuilder b) {
       STATIC,
       ABSTRACT,
       FINAL));
+
+    b.rule(REFERENCE).is(AMPERSAND);
+    b.rule(FUNCTION_DECLARATION).is(FUNCTION, b.optional(REFERENCE), IDENTIFIER,
+      LPARENTHESIS, b.optional(PARAMETER_LIST), RPARENTHESIS, BLOCK);
+
   }
 
   public static void statement(LexerlessGrammarBuilder b) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This method has 164 lines, which is greater than the 100 lines authorized. Split it into smaller methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS138)